tests/core/bytes/test_core_bytes.odin

```diff
@@ -0,0 +1,85 @@
+package test_core_bytes
+
+import "core:bytes"
+import "core:slice"
+import "core:testing"
+
+@test
+test_index_byte_sanity :: proc(t: ^testing.T) {
+	// We must be able to find the byte at the correct index.
+	data := make([]u8, 64)
```

Comment:
I'd make this dependent on `SIMD_SCAN_WIDTH` (after making it a public constant, or duplicating the 8 * size_of(uintptr) thing):
```cpp
SIMD_SCAN_WIDTH :: 8 * size_of(uintptr)
INDEX_MAX :: SIMD_SCAN_WIDTH - 1

data := make([]u8, SIMD_SCAN_WIDTH * 2)
defer delete(data)
slice.fill(data, '-')

for offset in 0..<INDEX_MAX {
	for idx in 0..<INDEX_MAX {
```